src/bin_node/node_storage_command.ml

```diff
@@ -0,0 +1,149 @@
+        term = Index.Integrity_check.term;
+      };
+      {
+        name = "check-self-contained";
+        description =
+          "Check that the upper layer of the store is self-contained.";
+        term = Pack.Check_self_contained.term;
+      } ]
+
+  let dispatch_subcommand = function
+    | None ->
+        `Help (`Auto, Some term_name)
+    | Some n -> (
+      match List.find_opt (fun {name; _} -> name = n) terms with
+      | None ->
+          let msg =
+            let pp_ul = Fmt.(list ~sep:cut (const string "- " ++ string)) in
+            terms
+            |> List.map (fun {name; _} -> name)
+            |> Fmt.str
+                 "@[<v 0>Unrecognized command: %s@,\
+                  @,\
+                  Available commands:@,\
+                  %a@,\
+                  @]"
+                 n
+                 pp_ul
+          in
+          `Error (false, msg)
+      | Some command -> (
+          let (binary_name, argv) =
+            (* Get remaining arguments for subcommand evaluation *)
+            ( Sys.argv.(0),
+              Array.init
+                (Array.length Sys.argv - 2)
+                (function 0 -> Sys.argv.(0) | i -> Sys.argv.(i + 2)) )
+          in
+          let noop_formatter =
+            Format.make_formatter (fun _ _ _ -> ()) (fun () -> ())
+          in
+          Term.eval
+            ~argv
+            ~err:noop_formatter (* Defaults refer to non-existent help *)
+            ~catch:false (* Will be caught by parent [Term.eval_choice] *)
+            ( command.term,
+              Term.info (binary_name ^ " " ^ term_name ^ " " ^ command.name) )
+          |> function
+          | `Ok f ->
+              `Ok (f ())
+          | `Help | `Version ->
+              (* Parent term evaluation intercepts [--help] and [--version] *)
+              assert false
+          | `Error _ -> (
+              (* We want to display the usage information for the selected
+                 subcommand, but [Cmdliner] will only do this at evaluation
+                 time *)
+              Term.eval
+                ~argv:[|""; "--help=plain"|]
+                ( command.term,
+                  Term.info (binary_name ^ " " ^ term_name ^ " " ^ command.name)
+                )
+              |> function `Help -> `Ok () | _ -> assert false ) ) )
+
+  let term =
+    let subcommand =
+      (* NOTE: [Cmdliner] doesn't have a wildcard argument or mechanism for
+         deferring the parsing of arguments, so this term must explicitly
+         support any options required by the subcommands *)
+      Arg.(value @@ pos_all string [] (info ~docv:"COMMAND" []))
+      |> Term.(app (const List.hd_opt))
+    in
+    Term.(ret (const dispatch_subcommand $ subcommand))
+end
+
+module Manpage = struct
+  let command_description =
+    "The $(b,storage) command provides tools for introspecting and debugging \
+     the storage layer."
+
+  let commands =
+    [ `S Cmdliner.Manpage.s_commands;
```

Comment:
Just curious, I couldn't get the point of this from Cmdliner code (and I don't see any difference without it when running the cli). What does it do? :)

Reply:
This provides a set of subcommand explanations on the help page when running `./tezos-node storage`:

```
❭ tezos-node storage

...
COMMANDS
       check-self-contained
           Check that the upper layer of the store is self-contained.

       integrity-check-index
           Search the store for integrity faults and corruption.

       stat-index
           Print high-level statistics about the index store.

       stat-pack
           Print high-level statistics about the pack file.
...
```

which is unfortunately necessary since we can't rely on Cmdliner to generate this information when nesting subcommands in this way.